### PR TITLE
[IOTDB-3317] RatisConsensus fallback for StateMachine RuntimeException

### PR DESCRIPTION
**Pull request by SzyWilliam**

### CODE REVIEW

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/ApplicationStateMachineProxy.java [hunk 1]
```diff
@@ -116,8 +116,14 @@ public CompletableFuture<Message> applyTransaction(TransactionContext trx) {
...
 
-    TSStatus result = applicationStateMachine.write(applicationRequest);
-    Message ret = new ResponseMessage(result);
+    Message ret;
+    try {
+      TSStatus result = applicationStateMachine.write(applicationRequest);
+      ret = new ResponseMessage(result);
+    } catch (Exception rte) {
+      logger.error("application statemachine throws a runtime exception: ", rte);
+      ret = Message.valueOf("internal error. statemachine throws a runtime exception");
```
Comment (OneSizeFitsQuorum) [inline consensus/src/main/java/org/apache/iotdb/consensus/ratis/ApplicationStateMachineProxy.java:125]: include the exception message

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/ApplicationStateMachineProxy.java (hunk 1 above)
Comment (SzyWilliam) [reply consensus/src/main/java/org/apache/iotdb/consensus/ratis/ApplicationStateMachineProxy.java]: done

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**